src/actions/transactionActions.ts

```diff
@@ -118,25 +118,26 @@ export function fetchTransaction(hash: string, chain: string) {
     if (shouldFetchTransaction(getState(), hash)) {
       dispatch(requestTransaction(hash))
 
-      const transactionRequestPromises = [
-        fetch(`${GENERATE_BASE_URL()}/transaction/${hash}`),
-        fetch(`${GENERATE_BASE_URL()}/log/${hash}`),
-      ]
-
-      chain === 'neo2' &&
+      const [network] = getNetworkAndProtocol()
+      const transactionRequestPromises: Promise<any>[] = []
+      if (chain === 'neo3') {
+        transactionRequestPromises.push(NeoRest.transaction(hash, network))
+        transactionRequestPromises.push(NeoRest.log(hash, network))
+      } else {
+        transactionRequestPromises.push(
+          NeoLegacyREST.transaction(hash, network),
+        )
+        transactionRequestPromises.push(NeoLegacyREST.log(hash, network))
         transactionRequestPromises.push(
-          fetch(`${GENERATE_BASE_URL()}/transaction_abstracts/${hash}`),
+          NeoLegacyREST.transactionAbstracts(hash, network),
         )
+      }
 
       try {
         const responses = await Promise.all(transactionRequestPromises)
```

Comment:
nitpick: we can make this a little verbose with something like this:
```javascript
const responses = await Promise.all(
  chain === "neo3"
    ? [NeoRest.transaction(hash, network), NeoRest.log(hash, network)]
    : [
        NeoLegacyREST.transaction(hash, network),
        NeoLegacyREST.log(hash, network).NeoLegacyREST.transactionAbstracts(
          hash,
          network
        ),
      ]
);
```

Reply:
I couldn't feed this it directly into `Promise.all` due to a massive type error. Squeezing it into a separate variable did work. Still cleaner than before, thanks for the idea 👍